### PR TITLE
Converted "DartPad in Tutorials: Best Practices" PDF to markdown

### PR DESCRIPTION
**Pull request by galeyang**
Converted "DartPad in Tutorials: Best Practices" PDF to markdown.

Staging link: https://jt-flutter-website-2.firebaseapp.com/resources/dartpad-best-practices

TODO:

- [x] Review the content and collect feedback
- [x] Update survey wording
- [ ] Update link in existing articles
- [ ] Do potential link redirection

### CODE REVIEW

File: src/_data/side-nav.yml [hunk 1]
```diff
@@ -226,6 +226,8 @@
       permalink: /resources
     - title: Books
       permalink: /resources/books
+    - title: dartpad best practices
+      permalink: /resources/dartpad-best-practices
```
Comment (legalcodes) [inline src/_data/side-nav.yml]:
```suggestion
title: DartPad in tutorials&#58; best practices
```

Comment (legalcodes) [inline src/_data/side-nav.yml]:
```suggestion
title: DartPad in tutorials&#58; best practices
```

File: src/_data/side-nav.yml [hunk 2]
```diff
@@ -226,6 +226,8 @@
       permalink: /resources
     - title: Books
       permalink: /resources/books
+    - title: dartpad best practices
```
Comment (legalcodes) [inline src/_data/side-nav.yml]:
```suggestion
title: DartPad in tutorials&#58; best practices
```

File: src/_data/side-nav.yml [hunk 3]
```diff
@@ -226,6 +226,8 @@
       permalink: /resources
     - title: Books
       permalink: /resources/books
+title: DartPad in tutorials&#58; best practices
```
Comment (legalcodes) [inline src/_data/side-nav.yml]:
```suggestion
    - title: DartPad in tutorials&#58; best practices
```

**Comment by kwalrath** [comment]
starting now...

**Comment by kwalrath** [comment]
When do you want to update links, remove the PDF, and do a redirect?

**Comment by kwalrath** [comment]
We talked out of band, and I'll handle the rest of the dart.dev-specific tasks...

**Pull request merged**